bowtie/_cli.py

```diff
@@ -780,6 +780,20 @@ def _validation_results_table_in_markdown(
     return final_content
 
 
+def _get_latest_dialect_report(
+    ctx: click.Context,
+    _,
+    report: _report.Report | None,
+):
+    if report is None:
+        latest_dialect = sorted(Dialect.known(), reverse=True)[0]
```

Comment:
This is `max(Dialect.known())`.

Also I think you can probably get away with *just* returning the report (and therefore not needing a callback at all, just the `default`) because I think you can get the dialect from `report.metadata.dialect` if I remember right (I haven't double checked).

Reply:
> Also I think you can probably get away with just returning the report (and therefore not needing a callback at all, just the default) because I think you can get the dialect from report.metadata.dialect if I remember right (I haven't double checked).

Okay @Julian, I tried doing it as you suggested i.e. using `report.metadata.dialect` but it seems to me that during the tests, `envsonschema` and `always_valid`  fake implementations or any other for that matter (e.g. even a real one like `python-jsonschema`) are adding a `Draft 2020-12` default dialect from their own side (reference:- https://github.com/bowtie-json-schema/bowtie/blob/84b56f153711bc001fb31a06abc9082cda5a866f/bowtie/_cli.py#L942) . Because of this `report.metadata.dialect` will always hold a dialect so I will have to change the tests [here](https://github.com/bowtie-json-schema/bowtie/blob/84b56f153711bc001fb31a06abc9082cda5a866f/tests/test_integration.py#L2259) to also include a new dialect line. Is it fine? I was doing this inclusion of a dialect line over here just so that if someone does just `bowtie statistics` then he/she knows that it was for the latest `draft2020` dialect report of the website and not just some random numbers on the console.

Reply:
> Oh, now I see/remember why I didn't add --dialect, it makes no sense when providing a report. Specifically cat "somereport | bowtie statistics --dialect draft7 makes no sense, the report is in whatever dialect the report is in.

Ofcourse I think you saying this will still hold true if the report is in whatever dialect it is (and that one will then appear in the `statistics` output) just that if None found then the implementations will assume / default to `Draft 2020-12` dialect.

Reply:
Also I think you can probably get away with just returning the report (and therefore not needing a callback at all, just the default) because I think you can get the dialect from report.metadata.dialect if I remember right (I haven't double checked).

Okay @Julian, I tried doing it as you suggested i.e. using report.metadata.dialect and it works 👍. However, I just got one query, is there a way we can print/echo on the console that if someone just does bowtie statistics then whatever numbers he/she is seeing on the console are of the latest report of the latest dialect from the website? or should we just leave it be as a helper text on the command?

Reply:
> is there a way we can print/echo on the console that if someone just does bowtie statistics then whatever numbers he/she is seeing on the console are of the latest report of the latest dialect from the website? 

I think I like what you were doing here -- you added a column that showed what dialect the report was for, I think you can also add one for the start time of the report and then that serves the purpose you mention.

Maybe we show a header that contains the report date and dialect, and then in the body we show the actual statistics.

Reply:
> Maybe we show a header that contains the report date and dialect, and then in the body we show the actual statistics.

for all formats we show a header ? (json, pretty, markdown)

Reply:
> Maybe we show a header that contains the report date and dialect, and then in the body we show the actual statistics.

@Julian How do we make changes in the tests for this 😅 ?

Reply:
I tried doing `Report.from_serialized(run_stdout)` in the tests to get the report date and dialect from the metadata (for assertion purposes) but I am getting a `JSONDecodeError` 😅.

Reply:
`Report.from_serialized` is what you use if you have a report serialized somewhere and want to deserialize it -- we don't have that here, stdout will contain some simple statistics and a small header so you can just assert directly against it I think -- or for the JSON output it should be a JSON output and you assert against the deserialized JSON. Does that help?

Reply:
No I mean the problem is I was trying to do `Report.from_serialized` because in the header we have report date + dialect. So how do we assert against the report date ?

Reply:
I again don't see what that function has to do with anything, your output is distinctly not a report.

Are you asking how to handle the fact that the date will not be some fixed time? The easiest way to handle that is probably to parse it manually using the datetime module and assert it's within 5 minutes of `now() `say, and then to just treat it as correct.

Reply:
Yeah sorry I wasn’t able to phrase my question correctly but yeah that’s what I meant. Thanks!